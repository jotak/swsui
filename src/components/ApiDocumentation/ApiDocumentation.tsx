// Only REST is supported for now, we're working on GraphQL !
//
// For REST, we use the excellent https://github.com/mrin9/RapiDoc
// https://github.com/mrin9/RapiDoc/blob/master/LICENSE.txt
//
// For now the 'allow-try' is set to false, but a PR will follow
// with an optional proxy to run requests against the service

import * as React from 'react';
import 'rapidoc/dist/rapidoc-min.js';

/* eslint-disable */
declare global {
  namespace JSX {
    interface IntrinsicElements {
      'rapi-doc': any;
    }
  }
}
/* eslint-enable */

interface ApiDocumentationProps {
    apiType: string;
    baseUrl: string;
  };

export class ApiDocumentation extends React.Component<ApiDocumentationProps> {

  render() {
    return (
      <div>
        <rapi-doc
           spec-url={this.props.baseUrl + '/apispec'}
           show-header="false"
           show-info="false"
           allow-authentication="false"
           allow-try="false"
           layout="column"
        />
     </div>
    );
  }
}
