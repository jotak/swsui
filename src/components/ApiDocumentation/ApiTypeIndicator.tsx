import * as React from 'react';
import { style } from 'typestyle';

import apiGrpcIcon from '../../assets/img/api-grpc-logo.svg';
import apiRestIcon from '../../assets/img/api-rest-logo.svg';
import apiGraphqlIcon from '../../assets/img/api-graphql-logo.svg';

interface Props {
  apiType: string;
}

const nameToSource = new Map<string, string>([
  ['grpc', apiGrpcIcon],
  ['rest', apiRestIcon],
  ['graphql', apiGraphqlIcon],
]);

const iconStyle = style({
  marginTop: -2,
  marginRight: 6,
  width: 30
});

export class ApiTypeIndicator extends React.Component<Props> {

  render() {
    return this.props.apiType ? this.renderIcon(this.props.apiType) : <span />;
  }

  renderIcon(apiType: string) {
    let iconToRender = nameToSource.get(this.props.apiType);
    return iconToRender ? (
      <img
        className={iconStyle}
        src={iconToRender}
        alt={apiType}
      />
    ) : (
      <span />
    )
  }

}
