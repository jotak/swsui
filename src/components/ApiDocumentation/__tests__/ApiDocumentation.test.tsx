import * as React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { ApiDocumentation } from '../ApiDocumentation';

describe('ApiDocumentation', () => {
  it('renders when not known', () => {
    let wrapper = shallow(<ApiDocumentation apiType="rest" baseUrl="aaa" />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
