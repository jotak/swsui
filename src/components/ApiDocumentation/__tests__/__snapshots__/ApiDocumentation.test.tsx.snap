// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApiDocumentation renders when not known 1`] = `
<div>
  <rapi-doc
    allow-authentication="false"
    allow-try="false"
    layout="column"
    show-header="false"
    show-info="false"
    spec-url="/api/namespaces/undefined/services/undefined/apispec"
  />
</div>
`;
